actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -327,11 +329,10 @@ def button_to(name = nil, options = nil, html_options = nil, &block)
         html_options = convert_options_to_data_attributes(options, html_options)
         html_options["type"] = "submit"
 
-        button = if block_given?
-          content_tag("button", html_options, &block)
+        button = if block_given? || button_to_generates_button_tag
+          content_tag "button", name || url, html_options, &block
         else
-          html_options["value"] = name || url
-          tag("input", html_options)
+          tag "input", html_options.with_defaults(value: name || url)
```

Comment:
Can we revert the code to the previous implementation? It is better if we don't create a new hash instance and it is clear what is happening than the `with_defaults` method. Also keep the parentheses.

Reply:
Done!